heareval/tasks/pipeline.py

```diff
@@ -291,6 +293,11 @@ def split_train_test_val(self, metadata: pd.DataFrame):
 
     def run(self):
         process_metadata = self.get_process_metadata()
+        # Check if one slug is associated with only one relpath
+        assert (
+            process_metadata.groupby("slug")["relpath"].nunique() == 1
+        ).all(), "There are duplicate slugs for one audio file"
+
```

Comment:
I think this confusing that this isn't where slugify_file_name is called and I worry that overriding methods might skip this check or do it wrong.

Also I think this assert is wrong. We don't want different slugs for one relpath to have nunique 1.

We want relpath.nunique == slug.nunique for ALL relpaths.

And we want that if we map from all replath to all slug, that there is a one to one mapping between slug and relpath

Reply:
run is always called and is what dispatches the rest of the pipelines. slugify_file_name is a helper that can be used in deriving classes. So this is the correct place for the check.

We definitely want relpath.nunique == slug.nunique.

Reply:
Yes. `run()` is never overridden, so the test can be here.
For the assert, cases - 
1. `One slug for one relpath` -- the bug which we were having is one slug for two relpath(relpath with -6 as well as +6 having the same slug), groupby by slug and see if one relpath is associated with one slug - this is done in the `assert` statement.
2. `One relpath for one slug` -- this is always true, as the slug function is deterministic.
3. `relpath.nunique() == slug.nunique()`, automatically holds if the above two holds.

So, by the above reasoning, the current assertion is sufficient. But please let me know if I miss some case.